### DISCUSSION TITLE
how to use the different learning rate in different module in lightning?

### DISCUSSION THREAD

**Discussion by huangfu170** [Lightning-AI/pytorch-lightning#20174] [General]
I want to set different learning rate for different submodules like 1e-5 for pretrained model and 5e-5 for linear layers, I use the following codes, but the training result seems weird:
`    def get_parameters(self):
        no_decay = ["bias", "LayerNorm.weight"]

        bert_param_optimizer = list(self.bert.named_parameters())
        bert_param_optimizer.extend(list(self.bert_label.named_parameters()))
        # bert_param_optimizer.extend(list(model.bert_cp.named_parameters()))
        # linear_param_optimizer = list(model.output_layer.named_parameters())
        linear_param_optimizer = list(self.GAT.named_parameters())
        # linear_param_optimizer.extend(list(model.project_gat.named_parameters()))
        linear_param_optimizer.extend(list(self.project_final.named_parameters()))
        linear_param_optimizer.extend(list(self.project_layer.named_parameters()))
        # linear_param_optimizer.extend(list(model.project_layer_cp.named_parameters()))
        # linear_param_optimizer.extend(list(model.project_layer_content.named_parameters()))
        linear_param_optimizer.extend(list(self.max_pool.named_parameters()))
        linear_param_optimizer.extend(list(self.scaled_att.named_parameters()))
        linear_param_optimizer.extend(list(self.text_cnn_cp.named_parameters()))
        linear_param_optimizer.extend(list(self.text_cnn_kw.named_parameters()))
        linear_param_optimizer.extend(list(self.label_lstm.named_parameters()))
        
        optimizer_grouped_parameters = [
            {'params': [p for n, p in bert_param_optimizer if not any(nd in n for nd in no_decay)],
            'weight_decay': cf.weight_decay, 'lr': cf.lr},
            {'params': [p for n, p in bert_param_optimizer if any(nd in n for nd in no_decay)], 'weight_decay': 0.0,
            'lr': cf.lr},
    
            {'params': [p for n, p in linear_param_optimizer if not any(nd in n for nd in no_decay)],
            'weight_decay': cf.weight_decay, 'lr': cf.line_lr},
            {'params': [p for n, p in linear_param_optimizer if any(nd in n for nd in no_decay)], 'weight_decay': 0.0,
            'lr': cf.line_lr},
        ]
        parameters=optimizer_grouped_parameters
        return parameters
    def configure_optimizers(self):
        # parameters=self.get_parameters()
        # parameters=self.parameters()
        optimizer=torch.optim.Adam(self.parameters(),lr=cf.lr)
        return {
            "optimizer":optimizer,
            # "scheduler":torch.optim.lr_scheduler.LinearLR(optimizer,total_iters=5)
        }`
